ignorelist.txt

```diff
@@ -0,0 +1,7 @@
+[unsigned-integer-overflow]
+src:/opt/irods-externals/boost-libcxx1.81.0-1/include/boost/beast/http/impl/field.ipp
+src:/opt/irods-externals/boost-libcxx1.81.0-1/include/boost/beast/core/impl/string.ipp
+src:/opt/irods-externals/boost-libcxx1.81.0-1/include/boost/beast/core/detail/string.hpp
+
+[implicit-conversion]
+src:/opt/irods-externals/boost-libcxx1.81.0-1/include/boost/asio/detail/impl/signal_set_service.ipp
```

Comment:
Consider replacing the contents of this file with just
```
src:/opt/irods-externals/*
```
That way all third-party headers are ignored.

If the specificity/granularity here is desired, however, consider instead replacing all instances of `boost-libcxx1.81.0-1` with `boost*`, so that this file doesn't need to be updated if our boost package is updated or if the non-libcxx version is used.

Also, consider a less generic name for this file, such as `ub-sanitizer-ignorelist.txt`

Reply:
> `ub-sanitizer-ignorelist.txt`

yes please

Reply:
If I recall correctly, ignore lists can be used for asan too. Do we wish to exclude /opt/irods-externals/* from asan too?

This would also move the `-fsanitize-ignorelist` to the shared sanitizer options of the cmake config.

Reply:
Not right now. Put that in an issue and we'll handle it later/separately.

Reply:
Will do the following:

- Change ignore list name to `ub-sanitizer-ignorelist.txt`
- Have the ignores be more general (use `boost*`)

I think the general ignore (`src:/opt/irods-externals/*`) is something we should consider with the 'use ignore lists with asan' issue that'll be up in a moment..

Reply:
Does `boost*` and `src:/opt/irods-externals/*` produce different results in some way?

Reply:
By `boost*`, I mean to leave the ignore list virtually as is, just glob the boost build as in:
`src:/opt/irods-externals/boost*/include/boost/asio/detail/impl/signal_set_service.ipp`

A scoped `src:/opt/irods-externals/*` would ignore all externals for the given scope (e.g. `[implicit-conversion]`).
An unscoped `src:/opt/irods-externals/*` would ignore all externals for all sanitizers, including ASAN.

For why the stricter ignore vs. a more general version, I'm not too sure how this would play with header-only items, nor if this would cause false-negatives. I think false positives would be preferred, and in this case, the current false positives are ignored.

Reply:
I don't have a strong preference on the ignorelist at this time. We want to keep third party code out of the sanitizer report. As long as that's satisfied, we're good. We can always tweak things later if needed.

Pick the one that makes sense to you.

Reply:
Everything should be done here, resolving.